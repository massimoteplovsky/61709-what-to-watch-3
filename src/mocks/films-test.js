export const films = [
  {
    id: 1,
    title: `Terminator 2. Judgment day`,
    genre: `Action`,
    year: 1991,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: [],
    src: ``,
    runtime: 5,
    reviews: [
      {
        text: ``,
        author: ``,
        date: ``,
        rating: 7.5
      }
    ]
  },
  {
    id: 2,
    title: `1+1`,
    genre: `Drama`,
    year: 2012,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: [],
    src: ``,
    runtime: 4,
    reviews: [
      {
        text: ``,
        author: ``,
        date: ``,
        rating: 7.5
      }
    ]
  },
  {
    id: 3,
    title: `Joker`,
    genre: `Drama`,
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: [],
    src: ``,
    runtime: 4,
    reviews: [
      {
        text: ``,
        author: ``,
        date: ``,
        rating: 7.5
      }
    ]
  },
  {
    id: 4,
    title: `Once upon a time in Hollywood`,
    genre: `Drama`,
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: [],
    src: ``,
    runtime: 4,
    reviews: [
      {
        text: ``,
        author: ``,
        date: ``,
        rating: 7.5
      }
    ]
  },
  {
    id: 5,
    title: `Ford vs Ferrari`,
    genre: `Drama`,
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: [],
    src: ``,
    runtime: 7,
    reviews: [
      {
        text: ``,
        author: ``,
        date: ``,
        rating: 7.5
      }
    ]
  },
  {
    id: 6,
    title: `1917`,
    genre: `Drama`,
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: [],
    src: ``,
    runtime: 4,
    reviews: [
      {
        text: ``,
        author: ``,
        date: ``,
        rating: 7.5
      }
    ]
  },
  {
    id: 7,
    title: `Parasite`,
    genre: `Drama`,
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: [],
    src: ``,
    runtime: 4,
    reviews: [
      {
        text: ``,
        author: ``,
        date: ``,
        rating: 7.5
      }
    ]
  },
  {
    id: 8,
    title: `Irishman`,
    genre: `Drama`,
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: [],
    src: ``,
    runtime: 4,
    reviews: [
      {
        text: ``,
        author: ``,
        date: ``,
        rating: 7.5
      }
    ]
  }
];

export const film = {
  id: 1,
  title: `Terminator 2. Judgment day`,
  genre: `Action`,
  year: 1991,
  promoPoster: ``,
  poster: ``,
  cover: ``,
  description: ``,
  rating: 8.5,
  votes: 12000,
  director: ``,
  starring: [`Tim cook`],
  src: ``,
  runtime: 4,
  reviews: [
    {
      text: ``,
      author: ``,
      date: ``,
      rating: 7.5
    }
  ]
};

export const promoFilmInfo = {
  id: 1,
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  year: 2014,
  poster: ``,
  src: ``
};

