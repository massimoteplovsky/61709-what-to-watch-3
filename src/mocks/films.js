export default [
  {
    id: 1,
    title: `Terminator 2. Judgment day`,
    poster: `http://i4.ytimg.com/vi/HgV7-MJwUBw/mqdefault.jpg`
  },
  {
    id: 2,
    title: `1+1`,
    poster: `https://i.pinimg.com/736x/76/44/c0/7644c07844dd4cb28d2c289bfa6e0dec--the-intouchables-comedy-movies.jpg`
  },
  {
    id: 3,
    title: `Joker`,
    poster: `https://all.culture.ru/uploads/9b743d135092e110a8e55a3dc3ad5bc4_w375_h250.jpeg`
  },
  {
    id: 4,
    title: `Once upon a time in Hollywood`,
    poster: `https://avatars.mds.yandex.net/get-pdb/1945878/3281b62e-4539-4eb7-a26b-fa68f6572bea/s375`
  },
  {
    id: 5,
    title: `Ford vs Ferrari`,
    poster: `https://sun9-28.userapi.com/c206816/v206816832/1771c/3p_wj8sfMz8.jpg`
  },
  {
    id: 6,
    title: `1917`,
    poster: `https://sun3-13.userapi.com/c857724/v857724955/14e882/u5CFMvbvX3I.jpg`
  },
  {
    id: 7,
    title: `Parasite`,
    poster: `http://m.kino-teatr.ru/art/5395/pv_75950.jpg`
  },
  {
    id: 8,
    title: `Irishman`,
    poster: `https://sun9-31.userapi.com/c850608/v850608676/40eb4/ojB5App6exk.jpg`
  },
];
