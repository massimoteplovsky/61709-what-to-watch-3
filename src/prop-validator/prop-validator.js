import {
  exact,
  string,
  number,
  arrayOf,
  func,
  array,
  bool
} from "prop-types";

const FilmPropType = {
  PROMO_FILM_INFO: exact({
    title: string.isRequired,
    genre: string.isRequired,
    year: number.isRequired
  }),
  FILM_INFO: exact({
    id: number.isRequired,
    title: string.isRequired,
    genre: string.isRequired,
    year: number.isRequired,
    promoPoster: string.isRequired,
    poster: string.isRequired,
    cover: string.isRequired,
    description: string.isRequired,
    rating: number.isRequired,
    votes: number.isRequired,
    runtime: number.isRequired,
    director: string.isRequired,
    starring: array.isRequired,
    src: string.isRequired,
    reviews: arrayOf(
        exact({
          text: string.isRequired,
          author: string.isRequired,
          date: string.isRequired,
          rating: number.isRequired
        })
    ).isRequired
  }).isRequired,
  TITLE_CLICK: func.isRequired,
  CARD_MOUSE_ENTER: func.isRequired,
  CARD_MOUSE_LEAVE: func.isRequired,
  SHOW_MORE_FILMS: func.isRequired,
  FILM_COUNTER: number.isRequired,
  RESET_FILM_COUNTER: func.isRequired
};

const VideoPropType = {
  IS_PLAYING: bool.isRequired,
  SRC: string.isRequired,
  IS_MUTED: bool.isRequired,
  POSTER: string.isRequired,
  CHANGE_PLAYER_RUN_MODE: func.isRequired,
  RENDER_VIDEO_PLAYER: func.isRequired
};

const MovieTabs = {
  ACTIVE_ITEM_INDEX: number.isRequired,
  CHANGE_ACTIVE_ITEM: func.isRequired
};

const MovieGenre = {
  CHANGE_GENRE: func.isRequired,
  GENRE: string.isRequired,
};

export const PropValidator = Object.assign(
    FilmPropType,
    {FILMS: arrayOf(FilmPropType.FILM_INFO).isRequired},
    VideoPropType,
    MovieTabs,
    MovieGenre
);

