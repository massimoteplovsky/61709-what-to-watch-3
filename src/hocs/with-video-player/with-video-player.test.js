import React from "react";
import renderer from "react-test-renderer";
import {PropValidator} from "../../prop-validator/prop-validator";
import withVideoPlayer from "./with-video-player.js";
import {Provider} from "react-redux";
import configureMockStore from "redux-mock-store";
import {films} from "../../mocks/films-test";

const mockStore = configureMockStore([]);
let store = mockStore({
  films: {
    films
  }
});

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropValidator.CHILDREN
};

const MockComponentWrapped = withVideoPlayer(MockComponent);

it(`withVideoPlayer is rendered correctly`, () => {
  const tree = renderer.create((
    <Provider store={store}>
      <MockComponentWrapped
        film={``}
        isPlaying={false}
        isMuted={true}
        isPreviewMode={true}
      />
    </Provider>
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
