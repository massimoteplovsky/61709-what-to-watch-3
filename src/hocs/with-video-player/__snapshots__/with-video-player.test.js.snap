// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withVideoPlayer is rendered correctly 1`] = `
<div>
  <video
    height="175"
    muted={true}
    poster=""
    src=""
    style={
      Object {
        "objectFit": "cover",
      }
    }
    width="280"
  />
</div>
`;
